### ISSUE TITLE
Please remove this assertion 

### ISSUE THREAD

**Issue by ARozental** [pytorch/pytorch#57495]
https://github.com/pytorch/pytorch/blob/4cb534f92ef6f5b2ec99109b0329f93a859ae831/aten/src/ATen/native/cpu/IndexKernel.cpp#L430

This assertion looks like this:
`      if (!is_mask_bool) {
        TORCH_CHECK(mask_value <= static_cast<mask_t>(1), "Mask tensor can take 0 and 1 values only");
      }
`

but there are in fact several valid use cases where the attention mask should have floating numbers that are not 1 or 0, I have already encountered 2 such cases during my work:

**Comment by ngimel** [comment]
uint8 masks are deprecated, please use bool masks and you won't have this issue.
